ot/torch/lp.py

```diff
@@ -79,3 +78,69 @@ def ot_solve(a, b, M, num_iter_max=100000, log=False):
         G = emd(a2, b2, M2, log=False, numItermax=num_iter_max)
 
         return torch.from_numpy(G).type_as(M)
+
+
+def emd1D_loss(u_values, v_values, u_weights=None, v_weights=None, p=1, require_sort=True):
```

Comment:
name the function `ot_1d_loss` please in order to have a unified API. Also EMD is only for p=1 we have toi spop using it wrong in the ML comunity ;).

Reply:
Also the function should compte the ot loss along dimension 0 as samples are aways acessed along dim 0 in ML. Please add a parameter to the function dim=0 by default that allows to choos along which dimension the loss shoud be computed (basically replace all the -1 by dim).

Reply:
OK sounds fair, will change the location of the slicing. My point of view was more one of efficiency: the way the matrices are presented at the moment preserves contiguity so that it should be (theoretically) faster

Reply:
well with the dim parameter you can so it as you want, just saying the standard is usually this way ;)

Reply:
and saples are stored like this in all of POT (ot.dist for instance)